pkg/mounter/proxy/server/metrics.go

```diff
@@ -0,0 +1,332 @@
+	done := make(chan struct{})
+
+	go func() {
+		manager.wg.Wait()
+		close(done)
+	}()
+
+	select {
+	case <-done:
+		klog.V(4).InfoS("All monitoring goroutines finished")
+	case <-time.After(timeout):
+		klog.Warningf("WaitForAllMonitoring timed out after %v, some goroutines may still be running", timeout)
+	}
+}
+
+// checkMountPointStatus checks the current status of a mount point
+// Note:
+// When [strict] is set to true, any abnormality of the mount point will result in an error.
+// This may cause continuous polling failures if the mount point is manually unmounted but the Pod is not deleted.
+// If there is no specific requirement to manually unmount the mount point, [strict] should be set to false
+// to avoid unnecessary errors during monitoring.
+func (m *MountMonitor) checkMountPointStatus(target string, strict bool) (err error) {
+	// Check if it's a mount point
+	notMnt, err := m.raw.IsLikelyNotMountPoint(target)
+	if err != nil {
+		// case 1: mount point does not exist
+		if os.IsNotExist(err) {
+			if strict {
+				return fmt.Errorf("Mount point does not exist")
+			}
+			return nil
+		}
+		// case 2: mount point is corrupted
+		if mountutils.IsCorruptedMnt(err) {
+			return fmt.Errorf("Mountpoint is corrupted")
+		}
+		// case 3: failed to check mount point
+		return fmt.Errorf("Error checking mount point, error: %v", err)
+	}
+
+	// case 4: mount point is not mounted
+	if notMnt && strict {
+		return fmt.Errorf("Mountpoint is not mounted")
+	}
+	return nil
+}
+
+// checkAndUpdateMountStatus checks mount status and updates metrics
+func (m *MountMonitor) checkAndUpdateMountStatus() {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	// Only check if we're in monitoring state
+	if m.State != MonitorStateMonitoring {
+		return
+	}
+	old := m.readMetricsFiles(MetricsMountPointStatus)
+	// metrics file stores notHealthy as "1", healthy as "0"
+	oldUnhealthy := old == "1"
+	// Check mount point status
+	err := m.checkMountPointStatus(m.Target, false)
+	if err != nil {
+		klog.ErrorS(err, "Check mount point status with error", "target", m.Target)
+	} else {
+		klog.V(5).InfoS("Check mount point status successfully", "target", m.Target)
+	}
+	newUnhealthy := err != nil
+	if newUnhealthy != oldUnhealthy {
+		klog.Warningf("Mount point status changed. but the detailed reason was missing")
+		// Update status
+		m.updateMountPointMetrics(nil, nil, ptr.To(newUnhealthy), err)
+	}
+}
+
+func boolToBinaryString(b bool) string {
+	boolMap := map[bool]string{true: "1", false: "0"}
+	return boolMap[b]
+}
+
+// initializeMetrics initializes metrics files to empty/zero state
+func (m *MountMonitor) updateMountPointMetrics(
+	retryCont *int,
+	failoverCont *int,
+	notHealthy *bool,
+	lastExistError error) {
+	if retryCont != nil {
+		// Update mount_retry_count
+		retryFile := filepath.Join(m.MetricsPath, MetricsMountRetryCount)
+		if err := utils.WriteAndSyncFile(retryFile, []byte(strconv.Itoa(*retryCont)), 0644); err != nil {
```

Comment:
Do we need to sync? If the host goes down unexpectly, I think we can accept the metrics being lost.